### PR TITLE
Lehao - Undo command bug fixes

### PR DESCRIPTION
**Pull request by Li-Lehao**
The previous undo has such a bug: when deleting person and then undo, the client list of the affected appointment will not be refreshed (undo). This is because a state simply copies the reference of the schedule, and different schedules in different states actually refer to the same appointment and same client list. To solve this we need to create deep copy when initialize the state and get schedule of the state.

### CODE REVIEW

**Review by YaleChen299 — COMMENTED** [review]
waiting for a constructor level implementation

**Comment by Li-Lehao** [comment]
> waiting for a constructor level implementation

Hi I think in `Contacts` we are already using some constructors like this
```
public Contacts(ReadOnlyContacts toBeCopied) {
        this();
        resetData(toBeCopied);
    }
```
which creates a copy of the given `Contacts` instance.
However, this does not create a deep copy and the resulting new `Contacts` instance actually refer to the same set of `Person`.
We need a deep copy in this case but the function signature was taken.

**Review by YaleChen299 — COMMENTED** [review]
LGTM. I have tested with some commands. maybe just waiting for the filteredList predicates to be put into the state.

File: src/main/java/seedu/placebook/logic/commands/FindTagsCommand.java [hunk 1]
```diff
@@ -30,7 +30,8 @@ public FindTagsCommand(PersonHasTagsPredicate predicate) {
     public CommandResult execute(Model model, Ui ui) {
         requireNonNull(model);
         model.updateFilteredPersonList(predicate);
-        model.updateState();
+        model.updateState(
+                String.format(Messages.MESSAGE_PERSONS_LISTED_OVERVIEW, model.getFilteredPersonList().size()));
         return new CommandResult(
```
Comment (YaleChen299) [inline src/main/java/seedu/placebook/logic/commands/FindTagsCommand.java:35]: I took a few try. maybe for findTags, the message can be `undo command: findTags: 0 person listed`

**Review by YaleChen299 — COMMENTED** [review]
didnt find an issue when I test. Maybe the message can be have a `\n` after the undo command success

File: src/main/java/seedu/placebook/logic/commands/FindTagsCommand.java (hunk 1 above)
Comment (Li-Lehao) [reply src/main/java/seedu/placebook/logic/commands/FindTagsCommand.java:35]: Oh ya the MESSAGE_SUCCESS of `findTag` command does not contain its COMMAND_WORD. I will fix this.

**Review by YaleChen299 — APPROVED** [review]
LGTM

**Pull request merged**